### PR TITLE
Improve user docs on top of #124

### PR DESCRIPTION
**Pull request by sdruskat**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] The *Unreleased* section in CHANGELOG.md has been amended to reflect the changes in this PR
- [x] Build (`mvn verify`) was run locally and any changes were pushed
- [ ] The pull request is against the correct branch (`master` for bug fixes, `develop` for new functionality)

## Pull request type

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- Improve user documentation for the changes in #124
- Change label "Include **S**pans" to "Include **s**pans"
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Does this introduce new dependencies?

- [ ] Yes
- [x] No

If this introduces new dependencies:

- [ ] [Dependencies and citation templates](https://github.com/hexatomic/hexatomic/tree/develop/releng/templates) have been updated where necessary

## Other information

---

# Checklist for maintainers

## Releases

- [ ] Do **NOT** push the green merge button on GitHub.  
Follow the latest version of the [developer/maintainer docs](https://hexatomic.github.io/hexatomic/dev/) for making releases.
- [ ] License and citation information for dependencies are complete (each dependency has a folder in `/THIRD-PARTY/`).

### CODE REVIEW

File: docs/user/src/usage/graph-editor/console.md [hunk 1]
```diff
@@ -27,13 +27,18 @@ This command will result in 5 tokens: `[This] [is] [an] [example] [.]`.
 
 If you call `t` again, the new token will be appended to the end.
 E.g. calling `t Not .` will result in 7 tokens in total: `[This] [is] [an] [example] [.] [Not] [.]`.
-Note that the dot is not escaped in this example.
+Note that the dot is not escaped with `"` quotation marks in this example, and that `t Not.` would also work.
 
+Escaping punctuation with quotation marks is especially important in cases like `t I'm a test.`, which results in the
+creation of 5 tokens: `[I] [m] [a] [test] [.]`. To keep the apostrophe as a separate token, escape it as `t I"'"m ...`.
```
Comment (thomaskrause) [inline docs/user/src/usage/graph-editor/console.md]: I actually think this is a bug. `t I'm a test.` should either
- create `'` as token: `[I] ['] [m] [a] [test] [.]`
- make `'` part of the token `[I'm]`
- throw an error that this is an unsupported syntax if `'` is not escaped

Its clear we need to document corner cases, but maybe we need to make sure we define them consistently. Since `'` might be used for later syntactic purposes, this example could e.g. suggest to always escape non-basic latin characters characters like `'`.

File: docs/user/src/usage/graph-editor/console.md [hunk 2]
```diff
@@ -27,13 +27,18 @@ This command will result in 5 tokens: `[This] [is] [an] [example] [.]`.
...
 
 
-## Tokenize before (`tb`) and after (`ta`)
+
+## Tokenize before (`tb`) and after (`ta`) a given token
 
 Tokenize the given argument string and add the tokens to the annotation graph before or after a given reference token.
+Read the section [Identifying elements](#identifying-elements) to find out how to pass the command the respective reference token.
```
Comment (thomaskrause) [inline docs/user/src/usage/graph-editor/console.md]: It is good to have a more detailed explanation of how to identify elements in a section that also can be separately linked, but I would not encourage the user to follow this link yet.
The examples sub-sections shows examples for how to reference tokens but in a relevant environment of the "how do i tokenize" example. 
My suggestion would be to move the link to the detailed explanation after the examples, so that the reader first gets an overview via the examples and then can decide to look up other instructions or get an in-deep explanation of referencing nodes.

File: docs/user/src/usage/graph-editor/console.md [hunk 3]
```diff
@@ -53,11 +58,14 @@ ta #t1 is a
 
 ## New node: `n`
 
-The command `n` will create a new node and optionally add annotations and dominance relations to existing nodes.
-Each of its new annotation arguments has the form `name:value` or `namespace:name:value`.
+The command `n` will create a new node, and dominance relations between the new node and existing nodes.
+
+Additionally, it can be used to annotate the new node in the same command.
+To find out how to define annotations, read the section [Defining annotation](#defining-annotations).
```
Comment (thomaskrause) [inline docs/user/src/usage/graph-editor/console.md]: Again, I would argue to first show the examples where the definitions are shown in context and then link to the detailed and optional definition.

File: docs/user/src/usage/graph-editor/console.md [hunk 4]
```diff
@@ -135,10 +143,12 @@ a pos: #t1
...
 
 
-## Delete node: `d`
+## Delete elements: `d`
 
 Deletes any node or edge of the graph.
-Give the entities to delete as an argument.
+Give the elements to delete as an argument.
+
+See [Identifying elements](#identifying-elements) to learn how to address elements.
```
Comment (thomaskrause) [inline docs/user/src/usage/graph-editor/console.md]: Do we need to repeat this in every section? I understand that we also have to cover the use case of people who just follow a link to a section describing a single command, but this gets very repetitive. Maybe we could add a "See also" section behind the example where we just list and link the relevant in depth-sub-section without embedding it into the text work-flow.

File: docs/user/src/usage/graph-editor/console.md [hunk 5]
```diff
@@ -152,4 +162,19 @@ Deletes nodes "t1" and "t2".
 d #t4 -> #t3
 ```
 
-Deletes the pointing relation edge between "t4" and "t3".
+Deletes the pointing relation between "t4" and "t3".
+
+## Identifying elements
+
+Elements in the graph are identified by the identifier on the node/edge in the graph. In the example below, the tokens have
+the identifiers `sTok1`, `sTok2`, `sTok3`, and `t28`. Hexatomic uses single character identifier prefixes (here: `t` for "token") and consecutive numbers.
```
Comment (thomaskrause) [inline docs/user/src/usage/graph-editor/console.md]: The identifiers are arbitrary. Its neither true that all tokens created by hexatomic will always follow the `tNr` scheme and also tokens in Salt can be named differently than with the prefix `sTok`. Teaching that to the user is confusing. Tokens are currently only distinguished by their color and position in the graph, not by their name.

File: docs/user/src/usage/graph-editor/console.md (hunk 1 above)
Comment (sdruskat) [reply docs/user/src/usage/graph-editor/console.md]: Error is caught and piped to stdout (`line 1:3 token recognition error at: '''`), but not duly registered by the listener, as ANTLR still goes on with the rest of the parse.

Instead the error should break the parse hard and report to the user interface.

Comment (sdruskat) [reply docs/user/src/usage/graph-editor/console.md]: This is #148.

File: docs/user/src/usage/graph-editor/console.md (hunk 2 above)
Comment (sdruskat) [reply docs/user/src/usage/graph-editor/console.md]: Good enough, I'll change and commit.

File: docs/user/src/usage/graph-editor/console.md (hunk 1 above)
Comment (sdruskat) [reply docs/user/src/usage/graph-editor/console.md]: Resolved in 0599e0aa8ef3368f98edfd3c7a0cec1cec0c3eec

File: docs/user/src/usage/graph-editor/console.md (hunk 3 above)
Comment (sdruskat) [reply docs/user/src/usage/graph-editor/console.md]: Resolved in ea040bf92f77418bcbe3874e971e94537d48bed9

File: docs/user/src/usage/graph-editor/console.md (hunk 5 above)
Comment (sdruskat) [reply docs/user/src/usage/graph-editor/console.md]: Resolved in ff2af0d97c5d63050fe06e16653ef3c3721d468d

**Comment by sdruskat** [comment]
@thomaskrause Please re-review and merge if acceptable, cheers. :beers:

File: docs/user/src/usage/graph-editor/console.md [hunk 6]
```diff
@@ -29,16 +36,13 @@ If you call `t` again, the new token will be appended to the end.
 E.g. calling `t Not .` will result in 7 tokens in total: `[This] [is] [an] [example] [.] [Not] [.]`.
 Note that the dot is not escaped with `"` quotation marks in this example, and that `t Not.` would also work.
 
-Escaping punctuation with quotation marks is especially important in cases like `t I'm a test.`, which results in the
-creation of 5 tokens: `[I] [m] [a] [test] [.]`. To keep the apostrophe as a separate token, escape it as `t I"'"m ...`.
+Other than that, escaping punctuation with quotation marks is required for all non-alphabetical characters, to ensure correct tokenization, such as in `t I"'"m ...`.
```
Comment (thomaskrause) [inline docs/user/src/usage/graph-editor/console.md]: Even if not required by the syntax (at least when mixing escaped/non-escaped token strings), I would use spaces between the tokens in the example
```suggestion
Other than that, escaping punctuation with quotation marks is required for all non-alphabetical characters, to ensure correct tokenization, such as in `t I "'" m ...`.
```
